ThingIFSDK/ThingIFSDK/Gateway/GatewayAPI.swift

```diff
@@ -0,0 +1,529 @@
+                    dispatch_async(dispatch_get_main_queue()) {
+                        completionHandler(error)
+                    }
+                }
+            )
+            let operation = IoTRequestOperation(request: request)
+            operationQueue.addOperation(operation)
+        } catch(_) {
+            kiiSevereLog("ThingIFError.JSON_PARSE_ERROR")
+            completionHandler(ThingIFError.JSON_PARSE_ERROR)
+        }
+    }
+
+    /** Restore the Gateway.
+     This API can be used only for the Gateway App.
+
+     - Parameter completionHandler: A closure to be executed once finished. The closure takes 1 argument: an instance of ThingIFError when failed.
+     */
+    public func restore(
+        completionHandler: (ThingIFError?)-> Void
+        )
+    {
+        if !self.isLoggedIn() {
+            completionHandler(ThingIFError.USER_IS_NOT_LOGGED_IN)
+            return;
+        }
+
+        let requestURL = "\(self.gatewayAddress.absoluteString)/gateway-app/gateway/restore"
+
+        // generate header
+        let requestHeaderDict:Dictionary<String, String> = [
+            "authorization": "Bearer \(self.accessToken!)"
+        ]
+
+        // do request
+        let request = buildDefaultRequest(
+            HTTPMethod.POST,
+            urlString: requestURL,
+            requestHeaderDict: requestHeaderDict,
+            requestBodyData: nil,
+            completionHandler: { (response, error) -> Void in
+                dispatch_async(dispatch_get_main_queue()) {
+                    completionHandler(error)
+                }
+            }
+        )
+        let operation = IoTRequestOperation(request: request)
+        operationQueue.addOperation(operation)
+    }
+
+    /** Replace end-node by new vendorThingID for end node thingID.
+
+     - Parameter endNodeThingID: ID of the end-node assigned by Kii Cloud.
+     - Parameter endNodeVendorThingID: ID of the end-node assigned by End Node vendor.
+     - Parameter completionHandler: A closure to be executed once finished. The closure takes 1 argument: an instance of ThingIFError when failed.
+     */
+    public func replaceEndNode(
+        endNodeThingID: String,
+        endNodeVendorThingID: String,
+        completionHandler: (ThingIFError?)-> Void
+        )
+    {
+        if !self.isLoggedIn() {
+            completionHandler(ThingIFError.USER_IS_NOT_LOGGED_IN)
+            return;
+        }
+
+        if endNodeThingID.isEmpty || endNodeVendorThingID.isEmpty {
+            completionHandler(ThingIFError.UNSUPPORTED_ERROR)
+            return;
+        }
+
+        let requestURL = "\(self.gatewayAddress.absoluteString)/\(self.app.siteName)/apps/\(self.app.appID)/gateway/end-nodes/THING_ID:\(endNodeThingID)"
+
+        // generate header
+        let requestHeaderDict:Dictionary<String, String> = [
+            "authorization": "Bearer \(self.accessToken!)",
+            "Content-Type": "application/json"
```

Comment:
@yongpingchen @naoki-akimoto 
I think Content-Type is required because this REST API send request body as JSON format.
[this line](https://github.com/KiiPlatform/thing-if-iOSSDK/pull/111/files#diff-9d9749dd18a02a4c609c2a3598dbc510R373) makes JSON format data. After that that data is sent.

What do you think @yongpingchen ?